controller/src/lib.rs

```diff
@@ -757,27 +754,33 @@ impl Controller {
         // override of the `LPMode` pin.
         match mode {
             PowerMode::Off => {
-                self.assert_lpmode(modules).await?;
+                // We would technically like to assert `LPMode` here. However,
+                // we can't do that without unintentionally back-powering the
+                // modules themselves. For now we deassert `LPMode`, but see
+                // https://github.com/oxidecomputer/hardware-qsfp-x32/issues/47
+                // for the hardware issue and
+                // https://rfd.shared.oxide.computer/rfd/0244#_fpga_module_sequencing
+                // for a general discussion.
+                self.deassert_lpmode(modules).await?;
                 self.assert_reset(modules).await?;
                 self.disable_power(modules).await
             }
-            other => {
+            new_mode @ PowerMode::Low | new_mode @ PowerMode::High => {
```

Comment:
Nit-picking my own previous suggestion, I just realized that we already have this variable captured in `mode`, so we don't actually need `new_mode` at all here!
```suggestion
           PowerMode::Low | PowerMode::High => {
```

Reply:
Ha true enough! Thanks.